lib/filter-stream.js

```diff
@@ -168,39 +223,52 @@ function FilterStream( blockMap, options ) {
    */
   _transform( chunk, _, next ) {
 
-    this._buffers.push( chunk )
-    this._bytes += chunk.length
+    debug( 'position', this.position )
+    debug( 'chunk', chunk.length, chunk )
 
-    if( this._bytes < this.blockSize )
-      return next()
+    this.bytesRead += chunk.length
+    this.blocksRead += chunk.length / this.blockSize
 
-    var buffer = Buffer.concat( this._buffers )
-    var offset = 0
+    if( this.currentRange == null ) {
+      debug( 'no current range' )
+      return next()
+    }
 
-    while( buffer.length - offset >= this.blockSize ) {
-      this._pushBlock( buffer.slice( offset, offset + this.blockSize ) )
-      offset = offset + this.blockSize
+    if( !this._rangeInChunk( chunk ) ) {
+      debug( 'chunk:ignore' )
```

Comment:
The thing I don't understand is that, if we skip discarded blocks, how Etcher (or anything else using FilterStream) will make to write the no discarded blocks in the correct place?
Shouldn't FilterStream be replacing the discarded blocks with zero (or random) -filled buffers that are the same size?
Or is there some mechanism to skip to the correct position in the output?